src/SDWaveFile.cpp

```diff
@@ -224,10 +259,30 @@ void SDWaveFile::readHeader()
     _file.close();
     return;
   }
-
   struct WaveFileHeader header;
 
-  if (_file.read(&header, sizeof(header)) != sizeof(header)) {
+  headerSize = sizeof(struct WaveFileHeader) - sizeof(header.subChunk2);
+  //Read the first WAV header's chunk
+  if (_file.read(&header, headerSize) != headerSize) {
+    _file.close();
+    return;
+  }
+
```

Comment:
I think we need the check for `header.subChunk1.size != 16` here, like I mentioned below.

Reply:
ok after this piece of code make sense, i'll add